### PR TITLE
Improve code/schema validation before registering

### PR DESCRIPTION
**Pull request by jl-santana**
More info about this can be found on this [issue](https://github.com/near/queryapi/issues/483)

Changes on this PR:
- Validate code & schema before registering the indexer
   Logic: If formatting either the code or the schema fails, the `Publish` button is disabled. Only if type generation errors or no errors are detected, the `Publish` button will be enabled. More about this on this [discussion](https://github.com/near/queryapi/pull/480#issuecomment-1878884650) 

https://github.com/near/queryapi/assets/15988846/68f89cb4-f561-4e8a-9fa7-81c1a38e548c

Additionally 
- Created a reusable Modal with a global context to manage it, so we can trigger it from any component to display some info
- Updated schema with granular error types for improved clarity
- Created a custom error class to filter by type Error. We can add more fields on it if needed

### CODE REVIEW

File: frontend/src/constants/Strings.js [hunk 1]
```diff
@@ -1,5 +1,10 @@
...
-export const CODE_FORMATTING_ERROR = "There was an error while formatting your code. Please check the console for more details";
-export const SCHEMA_FORMATTING_ERROR = "Oh snap! We could not generate types for your SQL schema. Make sure it is proper SQL DDL.";
\ No newline at end of file
+//errors messages
+export const CODE_GENERAL_ERROR_MESSAGE = "There is an error in your code. Please check the console for more details.";
+export const CODE_FORMATTING_ERROR_MESSAGE = "There was an error while formatting your code. Please check the console for more details.";
+export const SCHEMA_TYPE_GENERATION_ERROR_MESSAGE = "There was an error while generating types for your SQL schema. Please ensure your schema is valid SQL DDL.";
+export const SCHEMA_FORMATTING_ERROR_MESSAGE = "There was an error while formatting your schema. Please, check the console for more details."
+export const INDEXER_REGISTER_TYPE_GENERATION_ERROR = "There was an error generating types from your schema, so the contextDB object cannot be created. Would you like to proceed with registering the Indexer despite this?";
```
Comment (roshaans) [inline frontend/src/constants/Strings.js:6]:
```suggestion
export const INDEXER_REGISTER_TYPE_GENERATION_ERROR = "There was an error generating types from your schema, so the context.db object cannot be created. Would you like to proceed with registering the Indexer despite this?";
```

File: frontend/src/components/Editor/EditorButtons.jsx [hunk 2]
```diff
@@ -117,33 +114,33 @@ const EditorButtons = ({
...
-                  <OverlayTrigger
-                    placement="bottom"
-                    overlay={<Tooltip>Delete Indexer</Tooltip>}
-                  >
-                    <Button
-                      size="sm"
-                      className="flex align-center"
-                      variant="danger"
-                      onClick={() => handleDeleteIndexer()}
+                    <OverlayTrigger
```
Comment (roshaans) [inline frontend/src/components/Editor/EditorButtons.jsx:117]: nit: Remove diff if no changes are being made

File: frontend/src/components/Editor/Editor.jsx [hunk 3]
```diff
@@ -333,96 +362,111 @@ const Editor = ({
...
+            Indexer Function could not be found. Are you sure this indexer exists?
+          </Alert>
+        )}
+        {(indexerDetails.code || isCreateNewIndexer) && <>
+          <EditorButtons
+            handleFormating={handleFormating}
+            handleCodeGen={handleCodeGen}
+            error={error}
+            executeIndexerFunction={executeIndexerFunction}
+            currentUserAccountId={currentUserAccountId}
```
Comment (roshaans) [inline frontend/src/components/Editor/Editor.jsx:390]: Remove the variables you removed from `EditorButtons` here. 

`currentUserAccountId`,

**Review by roshaans — APPROVED** [review]
Changes look good to me. 💯 

It would be nice though to clear out all errors that the user sees when they click the button to `Reset Changes`.  At the moment the errors stay.

File: frontend/src/components/Editor/Editor.jsx (hunk 3 above)
Comment (jl-santana) [reply frontend/src/components/Editor/Editor.jsx:390]: Done

**Pull request merged**